iroh-sync/src/sync.rs

```diff
@@ -110,7 +132,7 @@ pub struct Replica<S: ranger::Store<SignedEntry> + PublicKeyStore> {
 struct InnerReplica<S: ranger::Store<SignedEntry> + PublicKeyStore> {
     namespace: Namespace,
     peer: RwLock<Peer<SignedEntry, S>>,
-    on_insert_sender: RwLock<Option<flume::Sender<Event>>>,
+    subscribers: Mutex<Subscribers>,
```

Comment:
given that sending will happen much more than updating subscribers, might want to go for an `RwLock`

Reply:
See above, if we require explicit unsubscribes we can do this, if we want to auto-remove disconnected senders we need `&mut` for each send.